gemonlinedb/include/gem/onlinedb/AMC13Configuration.h

```diff
@@ -0,0 +1,220 @@
+#ifndef GEM_ONLINEDB_AMC13CONFIGURATION_H
+#define GEM_ONLINEDB_AMC13CONFIGURATION_H
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <memory>
+
+#include <xercesc/dom/DOMDocument.hpp>
+
+#include "gem/onlinedb/DataSet.h"
+#include "gem/onlinedb/PartReference.h"
+#include "gem/onlinedb/Run.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        // Forward declaration
+        class AMCConfiguration;
+
+        /**
+         * @brief Describes the configuration of an AMC13.
+         *
+         * @note Due to the presence of configuration variables of non-integral
+         *       types, this class doesn't realize the @c Configuration concept.
+         *       Serialization is achieved by specializing the relevant template
+         *       classes.
+         *
+         * @note Modifying objects of this class doesn't affect hardware.
+         */
+        class AMC13Configuration
+        {
+        private:
+            int m_FEDId;
+            bool m_enableLocalTTC;
+            std::string m_hostname;
+
+            std::vector<std::shared_ptr<AMCConfiguration>> m_AMCConfigs;
+
+        public:
+            /**
+             * @brief Compares two AMC configurations for equality.
+             */
+            bool operator== (const AMC13Configuration &other) const;
+
+            ////////////////////////////////////////////////////////////////////
+
+            /**
+             * @brief Retrieves the id of the attached FED.
+             */
+            int getFEDId() const { return m_FEDId; }
+
+            /**
+             * @brief Modifies the id of the attached FED.
+             */
+            void setFEDId(int id) { m_FEDId = id; }
+
+            /**
+             * @brief Retrieves whether local TTC is enabled.
+             */
+            int isLocalTTCEnabled() const { return m_enableLocalTTC; }
+
+            /**
+             * @brief Enables or disables local TTC.
+             */
+            void setLocalTTCEnabled(bool enable) { m_enableLocalTTC = enable; }
+
+            /**
+             * @brief Retrieves the hostname.
+             */
+            std::string getHostname() const { return m_hostname; }
+
+            /**
+             * @brief Modifies the hostname.
+             */
+            void setHostname(const std::string &hostname)
+            {
```

Comment:
In my usual style, I try to not go past column 80, but it's a soft limit that can extend to ~90 when I think that the resulting code is visually more clear.

When an inline function fits on one line, I use:
> Example of good one-line function in header:
> ```cpp
> int getMyVal() { return m_myVal; }
> ```

When the above would result in a long line (ie >80, because going to 90 doesn't improve clarity), I use the "full function" layout:
> Example of good function definition:
> ```cpp
> void myFunc(args)
> {
>     statement1;
>     statement2;
> }
> ```
When this option is still too long, I break the argument list:
```cpp
int doSomethingVeryLong(int thisIsTooLong,
                        bool somethingElse)
{
    return m_myVal;
}
```
If it still doesn't fit, arguments move to the next line:
```cpp
int doSomethingVeryVeryVeryLong(
    int thisIsTooLong, bool somethingElse)
{
    return m_myVal;
}
```
If absolutely needed, the return type can move to the previous line (but I tend to use typedefs and `auto f() ->` instead):
```cpp
SomeVeryVeryVeryLongTypeName
SomeClass::doSomethingVeryVeryVeryLong(
    int thisIsTooLong, bool somethingElse)
{
    return m_myVal;
}
```
And nested template can be broken for clarity (this style is taken from boost):
```cpp
std::pair<
    std::vector<int>,
    std::uint32_t
> SomeClass::doSomethingVeryVeryVeryLong(
    int thisIsTooLong, bool somethingElse)
{
    return m_myVal;
}
```

IMHO adding a third style would be inconsistent with the above. Or do you mean that instead of:
```cpp
int getMyVal() { return m_myVal; }
```
one should use:
```cpp
int getMyVal() { 
    return m_myVal; };
```
? (Note: I dislike the extraneous semicolon.)

Reply:
> In my usual style, I try to not go past column 80, but it's a soft limit that can extend to ~90 when I think that the resulting code is visually more clear.

OK, that's what I figured (happens to be one of the style rules that I abandoned early on because people aren't coding on low resolution screens anymore :-) ); for line length "style", I go for clarity over hard limits (and in a linter I had used at some point, I had arbitrarily chosen 120 characters as the "warning" level, but mostly if it fits in my split pane in `emacs`, I'm happy)

> ? (Note: I dislike the extraneous semicolon.)

In the style of the project, it's used for *any* statement (including function implementation) in a header file (while not fundamentally necessary in `c++`, it has historical precedent, which is why it appears in a very large number of projects)

> IMHO adding a third style would be inconsistent with the above. Or do you mean that instead of:
>
> ```c++
> int getMyVal() { return m_myVal; }
> ```
>
> one should use:
>
> ```c++
> int getMyVal() { 
>     return m_myVal; };
> ```

Yes, if splitting the line of a single line function definition in a header file, the second form should be used if the first can't.  When it comes to multiple line function definitions in the header file, I haven't rigorously adopted a style (I could look back through the project and see what I did, but generally, when things became too verbose, I moved them to a corresponding source file).

Reply:
Should be correct now.

> In the style of the project, it's used for *any* statement (including function implementation)

In C++, a [function definition](https://en.cppreference.com/w/cpp/language/function#Function_definition) is a [declaration](https://en.cppreference.com/w/cpp/language/declarations), not a [statement](https://en.cppreference.com/w/cpp/language/statements) ;-)